account_payment_order/models/account_payment_order.py

```diff
@@ -91,8 +91,16 @@ class AccountPaymentOrder(models.Model):
     move_ids = fields.One2many(
         'account.move', 'payment_order_id', string='Journal Entries',
         readonly=True)
+    move_count = fields.Integer(
+        compute='_compute_move_count', readonly=True, store=True,
+        string="# of Journal Entries")
     description = fields.Char()
 
+    @api.depends('move_ids')
+    def _compute_move_count(self):
+        for order in self:
+            order.move_count = len(order.move_ids)
```

Comment:
This can scale bad when lot of moves, as this triggers the read of the records. Better to use `read_group` with domain `[('id', 'in', order.move_ids)]` and get the count from one of the values of the returned dictionary.